server/services/store/sqlstore/board_insights.go

```diff
@@ -0,0 +1,169 @@
+package sqlstore
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+
+	"github.com/mattermost/focalboard/server/model"
+
+	sq "github.com/Masterminds/squirrel"
+
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+func (s *SQLStore) getTeamBoardsInsights(db sq.BaseRunner, duration string, channelIDs []string) ([]*model.BoardInsight, error) {
+	/*
+		Get top private, public boards, combine the list and filter the top 10. Note we can't limit 10 for subqueries.
+	*/
+
+	qb := s.getQueryBuilder(db)
+	publicBoards := qb.Select(`blocks.id, blocks.title,
+		count(blocks_history.id) as count, blocks_history.modified_by, blocks.created_by`).
+		From(s.tablePrefix + "blocks_history as blocks_history").
+		Join(s.tablePrefix + "blocks as blocks on blocks_history.root_id = blocks.id").
+		Where(sq.Gt{"blocks_history.update_at": s.durationSelector(duration)}).
+		Where(sq.Eq{"blocks_history.workspace_id": channelIDs}).
+		Where(sq.NotEq{"blocks_history.modified_by": "system"}).
+		Where(sq.Eq{"blocks.delete_at": 0}).
+		GroupBy("blocks.title, blocks.created_by, blocks.id, blocks_history.modified_by")
+
+	insightsQuery := qb.Select(fmt.Sprintf("id, title, sum(count) as activity_count, %s as active_users, created_by",
+		s.concatenationSelector("distinct modified_by", ","))).
+		FromSelect(publicBoards, "boards_and_blocks_history").
+		GroupBy("id, title, created_by").
+		OrderBy("activity_count desc").
+		Limit(10)
+
+	rows, err := insightsQuery.Query()
+
+	if err != nil {
+		s.logger.Debug(`Team insights query ERROR`, mlog.Err(err))
+		return nil, err
+	}
+	defer s.CloseRows(rows)
+
+	boardInsights, err := boardsInsightsFromRows(rows)
+	if err != nil {
+		s.logger.Debug(`Team insights parsing error`, mlog.Err(err))
+		return nil, err
+	}
+	boardInsights, err = populateIcons(s, db, boardInsights)
+	if err != nil {
+		s.logger.Debug(`Team insights icon populate error`, mlog.Err(err))
+		return nil, err
+	}
+
+	return boardInsights, nil
+}
+
+func (s *SQLStore) getUserBoardsInsights(db sq.BaseRunner, userID string,
+	duration string, channelIDs []string) ([]*model.BoardInsight, error) {
+	/**
+	Get top 10 private, public boards, combine the list and filter the top 10
+	*/
+	qb := s.getQueryBuilder(db)
+	publicBoards := qb.Select(`blocks.id, blocks.title,
+		count(blocks_history.id) as count, blocks_history.modified_by, blocks.created_by`).
+		From(s.tablePrefix + "blocks_history as blocks_history").
+		Join(s.tablePrefix + "blocks as blocks on blocks_history.root_id = blocks.id").
+		Where(sq.Gt{"blocks_history.update_at": s.durationSelector(duration)}).
+		Where(sq.Eq{"blocks_history.workspace_id": channelIDs}).
+		Where(sq.NotEq{"blocks_history.modified_by": "system"}).
+		Where(sq.Eq{"blocks.delete_at": 0}).
+		GroupBy("blocks.title, blocks.created_by, blocks.id, blocks_history.modified_by")
+
+	userInsightsQuery := qb.Select("*").FromSelect(qb.Select(fmt.Sprintf("id, title, sum(count) as activity_count, %s as active_users, created_by",
```

Comment:
Why the `fmt.Sprintf` here?  Select supports an arbitrary number of string args.